### PR TITLE
Fix the flakiness in secrets extension test

### LINKED ISSUE #17501: [Flaky test] SecretsAreLoadedFromKeyVault
```

Error message
String lengths are both 1. Strings differ at index 0.
Expected: "2"
But was:  "1"
-----------^


Stack trace
   at Azure.Extensions.AspNetCore.Configuration.Secrets.Tests.ConfigurationSecretsFunctionalTests.SecretsAreLoadedFromKeyVault() in D:\a\1\s\sdk\extensions\Azure.Extensions.AspNetCore.Configuration.Secrets\tests\ConfigurationSecretsFunctionalTests.cs:line 39

```

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/17501

### CODE REVIEW

File: sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/ConfigurationSecretsFunctionalTests.cs [hunk 1]
```diff
@@ -33,6 +33,8 @@ public async Task SecretsAreLoadedFromKeyVault()
             Assert.AreEqual("2", configuration["TestSecret2"]);
             Assert.AreEqual("3", configuration["Nested:TestSecret3"]);
 
+            // KeyVault time resolution is 1sec we can't detect a change faster than that
```
Comment (Mohit-Chakraborty) [inline sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/ConfigurationSecretsFunctionalTests.cs:36]: From the comments, it seems like we would want to add the delay after updating the value and before checking for the new value. So, between `SetSecretAsync()` and `Reload()`.
Or, do you want to say the client does not allow updates to the same key with less than 1 second of granularity?

Comment (pakrym) [reply sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/ConfigurationSecretsFunctionalTests.cs:36]: The KeyVault stores `updated` property on the secret with 1 sec fidelity. So if the key is updated twice during the same second we won't be able to detect the change. That's why the delay is between the first `SetSecretAsync` and the second `SetSecretAsync`.

Comment (Mohit-Chakraborty) [reply sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/ConfigurationSecretsFunctionalTests.cs:36]: Ah, that's interesting. May have been better then for the API to allow clients to set `updated` so they can avoid this gotcha.

**Review by Mohit-Chakraborty — APPROVED** [review]

File: sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/ConfigurationSecretsFunctionalTests.cs (hunk 1 above)
Comment (pakrym) [reply sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/ConfigurationSecretsFunctionalTests.cs:36]: I think KeyVault service knows the exact time but for some reason, they decided to only store it to 1-second precision.

Even if there was a way to pass the exact value they won't store it.

**Review by toddrandy79 — COMMENTED** [review]
I have no clue and I'm starving and broken. What am I going to do now

**Pull request merged**